tools/unit_testing/run_tests.py

```diff
@@ -1,442 +1,329 @@
-
-#=================================================
-# Local modules.
-#=================================================
-
-this_script_dir = os.path.dirname(os.path.abspath(sys.argv[0]))
-
-python_dir = search_paths(
-    "printer.py",
-    [os.path.abspath("."),
-     os.path.abspath("python"),
-     this_script_dir,
-     os.path.join(this_script_dir, "python"),
-     ]
-    )
-
-sys.path.append(python_dir)
+logger = logging.getLogger(__name__)
 
-from machine_setup import MachineCompilerSettings
-from printer import Printer
-from xml_test_list import *
-
-#=================================================
-# Parse options and set up output Printer.
-#=================================================
-
-parser = OptionParser(
-    usage="%prog [options]",
+def parse_command_line(args):
+    """Command line argument parser for configure."""
     description="""Within build_directory (--build-dir), runs cmake on test
 specification directories (from --test-spec-dir or --xml-test-list), then
 builds and runs the tests defined via CMake."""
-    )
-parser.add_option(
-    "--build-dir", dest="build_dir", default=".",
-    help="""Directory where tests are built and run. Will be created if it
-does not exist."""
-    )
-parser.add_option(
-    "--build-type", dest="build_type", default="CESM_DEBUG",
-    help="""Value defined for CMAKE_BUILD_TYPE."""
-    )
-parser.add_option(
-    "--cesm-cmake", dest="cesm_cmake_dir",
-    help="""Location of CESM CMake modules.
+    parser = argparse.ArgumentParser(description=description)
 
-Usually this option is unnecessary, because this script can autodetect this
-location."""
-    )
-parser.add_option(
-    "--cesm-root", dest="cesm_root_dir",
-    help="""Location of CESM root directory.
+    CIME.utils.setup_standard_logging_options(parser)
 
-Usually this option is unnecessary, because this script can autodetect this
-location. If this option is set, it changes the locations used to
-autodetect all other paths."""
-    )
-parser.add_option(
-    "--clean", dest="clean", action="store_true",
-    default=False,
-    help="""Clean build directory before building. Removes CMake cache and
-runs "make clean"."""
-    )
-parser.add_option(
-    "--cmake-args", dest="cmake_args",
-    help="""Additional arguments to pass to CMake."""
-    )
-parser.add_option(
-    "--color", dest="color", action="store_true",
-    default=sys.stdout.isatty(),
-    help="""Turn on colorized output."""
-    )
-parser.add_option(
-    "--no-color", dest="color", action="store_false",
-    help="""Turn off colorized output."""
-    )
-parser.add_option(
-    "--compiler", dest="compiler", default="gnu",
-    help="""Compiler vendor for build (gnu, ibm, intel, nag, or pgi).
+    parser.add_argument("--build-dir", default=".",
+                        help="""Directory where tests are built and run. Will be created if it does not exist."""
+                        )
+    parser.add_argument("--build-type", default="CIME_DEBUG",
```

Comment:
You changed the default from `CESM_DEBUG` to `CIME_DEBUG`. I'm happy with this rename, but I see some other references to `CESM_DEBUG`; should these be changed consistently?:

```
share/csm_share/test/unit/shr_infnan_test/CMakeLists.txt:11:if("${CMAKE_BUILD_TYPE}" MATCHES CESM_DEBUG)
share/csm_share/test/unit/shr_infnan_test/CMakeLists.txt:24:  # CESM_DEBUG turns on floating point trapping, causing checks in this
tools/unit_testing/README:85:                     "CESM_DEBUG" will build with CESM's DEBUG mode
tools/unit_testing/python/machine_setup.py:96:                     Currently only works for CESM/CESM_DEBUG builds.
```

I was a bit surprised that `CESM_DEBUG` / `CIME_DEBUG` doesn't appear elsewhere, which made me wonder if I'm missing something here.